mlflow/mleap.py

```diff
@@ -167,3 +169,31 @@ def add_to_model(mlflow_model, path, spark_model, sample_input):
                             mleap_version=mleap.version.__version__,
                             model_data=mleap_datapath_sub,
                             input_schema=mleap_schemapath_sub)
+
+
+def _get_mleap_schema(dataframe):
+    """
+    :param dataframe: A PySpark dataframe object
+
+    :return: The schema of the supplied dataframe, in MLeap format. This serialized object of type
+    `ml.combust.mleap.core.types.StructType`, represented as a JSON dictionary.
+    """
+    from pyspark.ml.util import _jvm
+    ReflectionUtil = _jvm().py4j.reflection.ReflectionUtil
+
+    # Convert the Spark dataframe's schema to an MLeap schema object
+    tc_clazz = ReflectionUtil.classForName("org.apache.spark.sql.mleap.TypeConverters$")
+    tc_inst = tc_clazz.getField("MODULE$").get(tc_clazz)
+    mleap_schema_struct = tc_inst.sparkSchemaToMleapSchema(dataframe._jdf)
+
+    # Obtain a JSON representation of the MLeap schema object
+    js_clazz = ReflectionUtil.classForName("ml.combust.mleap.json.JsonSupport$")
+    js_inst = js_clazz.getField("MODULE$").get(js_clazz)
+    mleap_schema_json = js_inst.MleapStructTypeFormat().write(mleap_schema_struct)
+    return json.loads(mleap_schema_json.toString())
+
+
+def _handle_py4j_error(readable_error_text):
```

Comment:
Add a bit of a comment on what this is doing to handle py4j errors at a high level (e.g., does it print something to stderr and then re-raise? Not quite sure what `princ_exc` does)